### PR TITLE
Fix crawling

### PR DESCRIPTION
**Pull request by BombasterDS**
## About the PR
- Fixed buckling bug
- Now you'll be able to shoot laying people if you're laying too. Also you'll hit every obstacles like tables or crates.
- Stand up do after time increased from 0.5 to 1.5. Do After now cancel on damage taken.
- Crawling speed decreased from 0.4 to 0.3

## Why / Balance
It was broken

## Media

https://github.com/user-attachments/assets/cbea477a-0933-4f20-ab56-cf9abc8b5e1d

**Changelog**
:cl:
- fix: Fixed crawling bug with crawling.
- tweak: Now players can shoot lying mobs without keeping cursor on them when they alive.
- tweak: StandUp doafter time increased from 0.5 to 1.5

## Summary by CodeRabbit

- **New Features**
	- Enhanced projectile targeting system to respond dynamically to entity states, ensuring only appropriate targets are activated.
	- Introduced more nuanced control over the entity's standing and lying down mechanics.
	- Added new table functionalities, including `BulletImpassable` and `RequireProjectileTarget`, affecting projectile interactions.

- **Bug Fixes**
	- Fixed issues with downed entities remaining active when they should not be.

- **Gameplay Adjustments**
	- Increased time to stand up from a lying position from 0.5 seconds to 1.5 seconds.
	- Reduced movement speed while in a lying down state from 0.4 to 0.3.
	- Implemented a more assertive handling of the standing state when transitioning to lying down.

### CODE REVIEW

**Comment by BombasterDS** [comment]
so, testing failing because i uncommented integration test that was commented.
It will fail always, this system is to hardcoded and they made it hardcoded more. Everything looks fine in game.

File: Content.IntegrationTests/Tests/Buckle/BuckleTest.cs [hunk 1]
```diff
@@ -313,9 +313,7 @@ await server.WaitAssertion(() =>
             await server.WaitAssertion(() =>
             {
                 // Still buckled
-                //Assert.That(buckle.Buckled); // goob edit
-                // he's not supposed to be buckled with the new falling down system
-                // do i just did this :trollface:
+                Assert.That(buckle.Buckled);
```
Comment (Piras314) [inline Content.IntegrationTests/Tests/Buckle/BuckleTest.cs:316]: either comment this out or make the test work by I guess increasing the WaitRunTicks(10) to something higher, commenting it is legitimate

**Review by Piras314 — APPROVED** [review]

**Comment by whateverusername0** [comment]
wasn't shooting above people the *point* of crawling?

**Comment by BombasterDS** [comment]
> wasn't shooting above people the _point_ of crawling?

nah. It's too OP and makes you invincible to range and melee.

**Comment by whateverusername0** [comment]
hiding under tables is a rat people thing

**Comment by whateverusername0** [comment]
with the decreased speed and get up time you should keep shooting over laying down people

**Comment by BombasterDS** [comment]
> hiding under tables is a rat people thing

Still it looks not alright for many people on server.

**Comment by whateverusername0** [comment]
because that's the main point of crawling

ngl not being able to hit someone like that is a skill issue

**Comment by BombasterDS** [comment]
Plus it makes game less dynamic. Literally people just lying down in hallway and shooting nukies through the all screen because they can't shoot them back

**Comment by whateverusername0** [comment]
better make people who are laying down be able to hit both

**Comment by BombasterDS** [comment]
> better make people who are laying down be able to hit both

Hit both + hit all tables on the way. I think i can do it.

**Comment by whateverusername0** [comment]
yeah this will make a lot more sense

**Comment by BombasterDS** [comment]
> yeah this will make a lot more sense

Well, that took more times that i expected

**Comment by JoeHammad1844** [comment]
the reasons crawling is OP for firefights are..
1. in ss13 if you clicked on someone, your bullets would track the target, this would be the same if they were laying down
2. due to a bug with your cursor not lining up where you think it is under recoil, its basically impossible to hit somebody who is lying down unless you turn off recoil yourself in the settings

Shooting people who are lying down while you aren't aiming at them removes pretty much the entire point of crawling and makes it hard to have a firing line where people in the front lie down so those in the back can shoot over you

rodents lying under tables and being FULL ON IMPOSSIBLE TO HIT can be solved by not allowing them to sneak under tables in the first place

**Comment by JoeHammad1844** [comment]
i do agree that if you are lying down, you shouldn't need to click directly on top of someone who is also crawling to hit them

**Comment by BombasterDS** [comment]
> the reasons crawling is OP for firefights are..
> 
> 1. in ss13 if you clicked on someone, your bullets would track the target, this would be the same if they were laying down
> 2. due to a bug with your cursor not lining up where you think it is under recoil, its basically impossible to hit somebody who is lying down unless you turn off recoil yourself in the settings
> 
> Shooting people who are lying down while you aren't aiming at them removes pretty much the entire point of crawling and makes it hard to have a firing line where people in the front lie down so those in the back can shoot over you
> 
> rodents lying under tables and being FULL ON IMPOSSIBLE TO HIT can be solved by not allowing them to sneak under tables in the first place

I fixed it already - when you laying down, you'll be able not only hit laying down people, but tables. crates and other stuff that should be ignored. So rodents will shoot all the tables instead enemies.

**Comment by JoeHammad1844** [comment]
> I fixed it already - when you laying down, you'll be able not only hit laying down people, but tables. crates and other stuff that should be ignored. So rodents will shoot all the tables instead enemies.

neat

**Comment by whateverusername0** [comment]
> I fixed it already - when you laying down, you'll be able not only hit laying down people, but tables. crates and other stuff that should be ignored. So rodents will shoot all the tables instead enemies.

mad
approved

**Review by whateverusername0 — APPROVED** [review]

**Pull request merged**